lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -102,17 +117,23 @@ static LogicalResult checkBody(linalg::LinalgOp linalgOp) {
 static FailureOr<SmallVector<Value>>
 getSlicedOperands(OpBuilder &builder, Location loc, ValueRange localIvs,
                   linalg::LinalgOp linalgOp, ValueRange valuesToUse) {
-  assert(linalgOp->getNumOperands() == 3 &&
-         "expect 3 input/output operands");
+  assert(linalgOp->getNumOperands() == 3 && "expect 3 input/output operands");
   assert(linalgOp.getDpsInputOperands().size() == 2 &&
          "expect 2 input operands");
 
   SmallVector<Value> slicedOperands;
   for (OpOperand *operand : linalgOp.getDpsInputOperands()) {
-    FailureOr<Value> slicedOperand = utils::getSliceOperand(
-        builder, operand, linalgOp, localIvs, valuesToUse, 3);
-    if (failed(slicedOperand))
-      return failure();
+    FailureOr<Value> slicedOperand;
+    AffineMap map = linalgOp.getMatchingIndexingMap(operand);
+    if (map.getNumResults() == 5) {
+      slicedOperand = utils::getSliceOperand(builder, operand, linalgOp,
+                                             localIvs, valuesToUse, 4);
+    } else {
```

Comment:
No error checking here?

Reply:
what kind of error checking do you have in mind? I am extracting a map from the affine map of each opoperand and have a check for opoperand 1 to have a map with5 results. Do you think this map ought to be checked for vnni blocking?

Reply:
I just meant:
```
  if (failure(slicedOperand))
    return failure;
```